juju_verify/verifiers/base.py

```diff
@@ -181,24 +181,24 @@ def verify(self, check: str) -> Result:
         :raises VerificationError: If check fails in unexpected manner or if
                                    list of self.units is empty
         """
-        preflight_results = aggregate_results(
-            self.check_affected_machines(),
-            self.check_has_sub_machines()
-        )
-        verify_action = self._action_map().get(check)
-        if verify_action is None:
+        if check not in self.supported_checks():
             raise NotImplementedError('Unsupported verification check "{}" for'
                                       ' charm {}'.format(check, self.NAME))
 
+        def verify_action() -> Result:
+            return self._action_map()[check](self)
```

Comment:
In the previous code,  `.get(check)` was used. That approach avoided a `KeyError` exception and returned a callable method. Now, this local function runs the method (e.g. `self.verify_shutdown()`). Is this the desired approach?

Please add a docstring with some information about the purpose of `verify_action` (although the name may look self explanatory).

Reply:
The `KeyError` never happen, because it's catches above.

The `_action_map()` will return map of function, however to call them you need to insert self as parameter, otherwise you will get:
```python
Verification failed: verify_shutdown() missing 1 required positional argument: 'self'
```
I will add a docstring.

Reply:
I added `get_verify_action`, which will return verification check as callable function without any arguments.